### ISSUE TITLE
Non-existing notification handlers are silently ignored

### ISSUE THREAD

**Issue by dagwieers** [ansible/ansible#14923]
##### Issue Type:

- Bug Report
##### Ansible Version:

```
v2.0.1.0
```
##### Ansible Configuration:

Nothing in particular.
##### Summary:

When you run ansible-playbook on a set of tasks with handlers, the existence of the handler is not checked, but those handlers are silently ignored. Even when the task induced a change.

Because of this, people may make mistakes in handlers but won't ever know that things aren't as they would expect. Ansible could test the existence of handler references and warn about them (when there's no change) and fail (when they are being called).

Silently ignoring notification handlers that should/could be run is problematic.
##### Steps To Reproduce:

```
- hosts: localhost
  gather_facts: no
  handlers:
    - name: foo-bar
      debug: msg='Handler foo-bar ran'
  tasks:
    - name: 'Use an incorrect handler name, ansible-playbook does not care'
      command: 'true'
      changed_when: no
      notify:
        - foo_bar

    - command: 'true'
      notify:
        - bar-foo

    - command: 'true'
      notify:
        - foo-bar                                                                                                   
```

##### Expected Results:

Failure report about notification handlers that do not exist.

This failure should happen also for handlers that are not to be used (i.e. because there was no change).
##### Actual Results:

```
PLAY ***************************************************************************

TASK [Use an incorrect handler name, ansible-playbook does not care] ***********
ok: [localhost]

TASK [command] *****************************************************************
changed: [localhost]

TASK [command] *****************************************************************
changed: [localhost]

RUNNING HANDLER [foo-bar] ******************************************************
ok: [localhost] => {
    "msg": "Handler foo-bar ran"
}

PLAY RECAP *********************************************************************
localhost                  : ok=4    changed=2    unreachable=0    failed=0   
```

The first task (no change) and second task (change) should both report some failure because the notification handler does not exist. Silently ignoring this is a recipe for disaster !

**Comment by bcoca** [comment]
consequence of #13507

**Comment by dagwieers** [comment]
@bcoca I noticed this #17902 and it seems configurable, so fixed in v2.2 ?

**Comment by bcoca** [comment]
yes, I thought i had closed this ... doing so now.
